src/audio/kpb.c

```diff
@@ -1244,8 +1245,11 @@ static enum task_state kpb_draining_task(void *arg)
 		 */
 			comp_cl_info(&comp_kpb, "kpb: update history_depth by %d",
 				     *rt_stream_update);
+			spin_lock_irq(&kpb->lock, flags);
 			history_depth += *rt_stream_update;
 			*rt_stream_update = 0;
+			spin_unlock_irq(&kpb->lock, flags);
+
```

Comment:
the only case where the kpb->lock is used is for the transition to the INIT_DRAINING state
here you are already draining, so how does the lock help?

Reply:
It disables interrupts which are a concern here because if they happen in between 'history_depth += *rt_stream_update;' and '*rt_stream_update = 0;' you lose the real time stream which happened when we were draining HB.